### PR TITLE
DBZ-8240 Use ifNotExists rule for unique key constraint in Mariadb

### PR DESCRIPTION
**Pull request by ani-sha**
closes https://issues.redhat.com/browse/DBZ-8240 and correctly indents the rules and comments
upstream - https://github.com/antlr/grammars-v4/pull/4250

### CODE REVIEW

**Comment by Naros** [comment]
@ani-sha is this change already in the upstream grammar? If not, can you send an upstream PR, too?

**Comment by jpechane** [comment]
BTW, there is something wrong with CI - MySQL tests are not executed

**Comment by Naros** [comment]
> BTW, there is something wrong with CI - MySQL tests are not executed

Looks like we need to add a `mariadb-ddl-parser-changed` file change stage

**Comment by ani-sha** [comment]
@Naros Should add `mariadb-ddl-parser-changed` in this PR?

**Comment by jpechane** [comment]
@ani-sha Yes please with `[ci]` prefix commit so we test both the PR and CI.

File: .github/workflows/debezium-workflow-pr.yml [hunk 1]
```diff
@@ -314,7 +315,7 @@ jobs:
     name: "MySQL ${{ matrix.version-mysql-server }} - ${{ matrix.profile }}"
     needs: [ check_style, file_changes ]
     runs-on: ubuntu-latest
-    if: ${{ needs.file_changes.outputs.common-changed == 'true' || needs.file_changes.outputs.mysql-changed == 'true' || needs.file_changes.outputs.mysql-ddl-parser-changed == 'true' || needs.file_changes.outputs.schema-generator-changed == 'true' }}
+    if: ${{ needs.file_changes.outputs.common-changed == 'true' || needs.file_changes.outputs.mysql-changed == 'true' || needs.file_changes.outputs.mysql-ddl-parser-changed == 'true' || needs.file_changes.outputs.mariadb-ddl-parser-changed == 'true' || needs.file_changes.outputs.schema-generator-changed == 'true' }}
```
Comment (Naros) [inline .github/workflows/debezium-workflow-pr.yml:318]: @ani-sha this isn't necessary, we shouldn't run the MySQL tests when we modify the MariaDB grammar.

Comment (ani-sha) [reply .github/workflows/debezium-workflow-pr.yml]: I see, I got confused that we want to run both the grammars together for all changes. Fixed it now.

**Comment by jpechane** [comment]
@ani-sha Applied, thanks!

**Pull request merged**